crates/optimism/txpool/src/interop.rs

```diff
@@ -0,0 +1,41 @@
+//! Additional support for pooled transactions with [`TransactionInterop`]
+
+/// Helper trait that allows attaching a [`TransactionInterop`].
+pub trait MaybeInteropTransaction {
+    /// Attach a [`TransactionInterop`].
+    fn set_interop(&mut self, conditional: TransactionInterop);
+
+    /// Get attached [`TransactionInterop`] if any.
+    fn interop(&self) -> Option<&TransactionInterop>;
+
+    /// Helper that sets the conditional and returns the instance again
+    fn with_interop(mut self, interop: TransactionInterop) -> Self
+    where
+        Self: Sized,
+    {
+        self.set_interop(interop);
+        self
+    }
+}
+
+/// Helper to keep track of cross transaction interop validity
+#[derive(Debug, Clone, Default, Eq, PartialEq)]
+pub struct TransactionInterop {
+    /// Time until tx if validated successfully.
+    /// If None - tx is not validated, tx should be automatically revalidated by interop tracker
+    /// if timeout - current block timestamp < 1min
+    // TODO: we should choose some concrete timing for revalidation, but 1min sound reasonable
+    pub timeout: Option<u64>,
```

Comment:
let's mention that this is the unix timestamp

Reply:
instead of `Option` let's just have two fields:
1. is_interop_tx: AtomicBool
2. timeout: AtomicU64
and skip the lock

Reply:
We could just assume that if timeout = 0 it's false (uninitialized) and we could contain all in single field

Reply:
both are possible, we can even use a single AtomicU64 with a magic for the is_interop

but I'm also fine with doing an rwlock for this in case we need to add more into the type besides tracking the timeout.

Reply:
timeout zero is synonymous with "time to ask supervisor again", if that is semantically identical to "hasn't been validated", then sure, we can just have the one field